src/App.jsx

```diff
@@ -14,33 +15,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function sortByAlphabetically(copyAlphabeticallyArray) {
+  const sortAlphabeticallyArray = [...copyAlphabeticallyArray];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  return sortAlphabeticallyArray.sort((good1, good2) =>
+    good1.localeCompare(good2),
+  );
+}
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+function sortByLength(copyLengthArray) {
+  const sortLengthArray = [...copyLengthArray];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  return sortLengthArray.sort((good1, good2) => good1.length - good2.length);
+}
+
+function reverseCopyArray(copyReverseArray) {
+  const arrayReverse = [...copyReverseArray];
+
+  return arrayReverse.reverse();
+}
+
+export function App() {
+  const [goodValue, setGoodValue] = useState([...goodsFromServer]);
+  const [flag, setFlag] = useState(false);
+  const [activeMethod, setActiveMethod] = useState(null);
+
+  const sortAlphabetically = () => {
+    const sortedGoods = sortByAlphabetically(goodValue);
+
+    setGoodValue(sortedGoods);
+    setActiveMethod('alphabetically');
+    setFlag(true);
+  };
+
+  const sortLength = () => {
+    const sortedGoods = sortByLength(goodValue);
+
+    setGoodValue(sortedGoods);
+    setActiveMethod('length');
+    setFlag(true);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const reverseArr = () => {
+    const reverseArray = reverseCopyArray(goodValue);
+
+    setGoodValue(reverseArray);
+    setActiveMethod('reverse');
+    setFlag(true);
+  };
```

Comment:
no needs to make such a similar logic. you should create one function that make this sorting
Example:
```
const visibleGoods = getGoodsPrepared(goodsFromServer, { sortType, isReversed });
```